### DISCUSSION TITLE
URL to call pause-, play-, etc button functions over LAN?

### DISCUSSION THREAD

**Discussion by Retrofreak** [rern/rAudio#1121] [Feature Requests]
Hi, is there a simple way to control (for e.g.) the pause-button per direct URL call? Then I could use my phone hit the pause button when someone calls me and later start the music again. Perhaps this could have to be done via a simple URL - but the buttons are queried completely via Javascript. I can't get it somehow...  Thanks in advance for a little explanation :-)

**Comment by cYCL157** [comment]
a predecessor project used a GET url that had the command as the query string, but that seems to have changed.  Using Fiddler I can see it's change a little but same concept:

```
POST http://raudio/cmd.php HTTP/1.1
Host: raudio
Connection: keep-alive
Content-Length: 58
Accept: */*
X-Requested-With: XMLHttpRequest
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Origin: http://raudio
Referer: http://raudio/
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9

cmd=sh&sh%5B%5D=cmd.sh&sh%5B%5D=mpcplayback&sh%5B%5D=pause
```

**Reply by omnisy** [reply]
Did you succeed in replicating a POST command? Do I read that data correctly: Qe are to send http POST to `http://raudio/cmd.php` with these key/value pairs?
```
cmd=sh
sh[]=cmd.sh
sh[]=mpcplayback
sh[]=pause
```

**Comment by Retrofreak** [comment]
Oh, sorry I did't follow this thread the last days. And quick and dirty I solved it like this: 


```
<?php
# fast and dirty
# /srv/http/mad-control.php
#
# mpc commands: mpc --help
# example calls:
# http://raudio1.domain.local/mad-control.php?c=pause
# http://raudio1.domain.local/mad-control.php?c=play
# http://raudio1.domain.local/mad-control.php?c=volume&p=50

 if (isset($_GET['c'])) {
  $mpc = '/usr/bin/mpc';
  $command = $mpc . ' ' . $_GET['c'];
  if (isset($_GET['p'])) {
   $command = $command . ' ' . $_GET['p'];
  }
 $output = shell_exec($command);
 }
?>
```

Not perfect, but works ;-)  I use this on my phone to pause music on calls.

**Comment by omnisy** [comment]
Thanks @Retrofreak I got it working now, both with GET and POST.
With GET (your solution) we first create a new .php on the pi to handle the GET calls, as you showed.
With POST (cYCL157's suggestion) we can without changing anything on the pi use a .html form with the `method="POST"` set.

One thing to note after reading
https://github.com/rern/rAudio-1/blob/main/srv/http/cmd.php
https://github.com/rern/rAudio-1/blob/main/srv/http/bash/cmd.sh
is that rAudio by default looks set up to allow any LAN user to not only access its web interface _but also_ with http POST to `cmd.php` run _any_ bash or exec command as root on the rAudio device! @rern I think that should be mentioned on the readme, since it can be an issue/risk depending on what LAN the rAudio device is deployed to.

**Reply by rern** [reply]
To prevent web interface from unauthorized access: `Features` > `Password Login` - enable

**Reply by omnisy** [reply]
Right, but the current readme does not describe that access to the web interface currently also means access to any bash/exec command as root on the pi device through http POST. That's worth stating explicitly. Is that wide power needed for some current rAudio functionality? An alternative would be to change cmd.php to only allow some whitelisted bash/exec actions, similar to how cmd.sh currently does it.
